README.yaml

```
@@ -18,30 +18,42 @@ badges:
+  - If you want the label items in a different order, you can specify that, too, with the `label_order` list.
+  - You can set a maximum length for the `id`, and the module will create a (probably) unique name that fits within that length.
+    (The module uses a portion of the MD5 hash of the full `id` to represent the missing part, so there remains a slight chance of name collision.)
   - You can control the letter case of the generated labels which make up the `id` using `var.label_value_case`.
-  - The labels are also exported as tags. You can control the case of the tag names (keys) using `var.label_key_case`.
+  - The labels are also exported as tags. You can control the case of the tag names (keys) for the labels using `var.label_key_case`.
+  Unlike the tags generated from the label inputs, tags passed in via the `tags` input are not modified.
+  
+  
+  There is an unforunate collision over the use of the label/key `name`. Cloud Posse uses `name` in this module
+  to represent the component, such as `eks` or `rds`. AWS uses a tag with the key `Name` to store the full human-friendly
+  identifier of the thing tagged, which this module outputs as `id`, not `name`. So when converting input labels
+  to tags, the value of the `Name` key is set to the module `id` output, and there is no tag corresponding to the
+  module `name` output.
 
   It's recommended to use one `terraform-null-label` module for every unique resource of a given resource type.
   For example, if you have 10 instances, there should be 10 different labels.
   However, if you have multiple different kinds of resources (e.g. instances, security groups, file systems, and elastic ips), then they can all share the same label assuming they are logically related.
 
-  All [Cloud Posse modules](https://github.com/cloudposse?utf8=%E2%9C%93&q=terraform-&type=&language=) use this module to ensure resources can be instantiated multiple times within an account and without conflict.
+  All [Cloud Posse Terraform modules](https://github.com/cloudposse?utf8=%E2%9C%93&q=terraform-&type=&language=) use this module to ensure resources can be instantiated multiple times within an account and without conflict.
 
+  Cloud Posse convention is to use the labels like this:
+  - `namespace`: A short (3-4 letters) abbreviation of the company name, to ensure globally unique IDs for things like S3 buckets
+  - `tenant`: Usually omitted. For resources dedicated to a customer or subsidiary but housed in the company's accounts, a short code to identify the customer
+  - `environment`: A [short abbreviation](https://github.com/cloudposse/terraform-aws-utils/#introduction) for the AWS region hosting the resource, or `gbl` for resources like IAM roles that have no region
+  - `stage`: The name or role of the account the resource is for, such as `prod` or `dev`
+  - `name`: The name of the component that owns the resources, such as `eks` or `rds`
```

```suggestion
  - `namespace`: A short (3-4 letters) abbreviation of the company name, to ensure globally unique IDs for things like S3 buckets.
  - `tenant`: Usually omitted. For resources dedicated to a customer or subsidiary but housed in the company's accounts, a short code to identify the customer.
  - `environment`: A [short abbreviation](https://github.com/cloudposse/terraform-aws-utils/#introduction) for the AWS region hosting the resource, or `gbl` for resources like IAM roles that have no region.
  - `stage`: The name or role of the account the resource is for, such as `prod` or `dev`.
  - `name`: The name of the component that owns the resources, such as `eks` or `rds`.
```

Punctuation (sometimes it's fine not to add periods in a list such as this if each item is a small blurb, however some of these items are more than one sentence).